schema_based_on_diagram.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE medical_histories(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  admitted_at TIMESTAMP,
+  patient_id INT REFERENCES patients(id),
+  status VARCHAR(20),
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE patients(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  name VARCHAR(20),
+  date_of_birth DATE,
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE invoices(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history__id INT REFERENCES medical_histories(id),
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE treatments(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  type VARCHAR(20),
+  name VARCHAR(20),
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE invoice_items(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT REFERENCES invoices(id),
+  treatment_id INT REFERENCES treatments(id),
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE treatments_medical_histories(
+  treatment_id INT REFERENCES treatments(id),
+  medical_history_id INT REFERENCES medical_histories(id),
+  PRIMARY KEY(treatment_id, medical_history_id)
+);
```

Comment:
- [ ] Good job creating all the tables but kindly be noticed that you should create indexes for all FOREIGN KEYS.
```sql
CREATE INDEX ON medical_histories (patient_id);
CREATE INDEX ON invoices (medical_history_id);
CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON invoice_items (treatment_id);
CREATE INDEX ON treatments_medical_histories (medical_history_id);
CREATE INDEX ON treatments_medical_histories (treatment_id);
```

Reply:
Thank you